technical-docs/designs/infra/managed-identity.md

```diff
@@ -0,0 +1,251 @@
+
+- User-assigned managed identity: The identity that's used for authorization in
+  the cluster.
+
+This document outlines a testing suite to support feature related support for
+managed identities in AKS using a proposed new Bedrock environment that
+leverages a modified cobalt project test harness in order for test pod identity
+within an AKS cluster using agile CI/CD and test validation.
+
+### Issues Addressed:
+
+1. [As an SRE, I want Enable MSI Support for aks-gitops module](https://github.com/microsoft/bedrock/issues/994)
+2. [As an Operator, I want automated testing validation for MSI verified within Bedrock](https://github.com/microsoft/bedrock/issues/1197)
+3. [As an operator, I want integration Tests tracking with junit logs from terratest](https://github.com/microsoft/bedrock/issues/867)
+4. [As an operator, I want to implement a managed service identity (via AAD Pod Identity) based secret handling strategy](https://github.com/microsoft/bedrock/issues/482)
+
+## 2. Out of Scope
+
+An existing pull request for Bedrock currently exists that enables MSI support
+for aks-gitops modules [#995](https://github.com/microsoft/bedrock/pull/995).
+This design document seeks to solely capture a terraform template and
+complementary test.
+
+The following are not included in this proposal:
+
+- Mocking for Terraform Unit Tests
+- Feature revert and Rollback from failed merges
+- Adjusting Cobalt Test Fixture support for current file organization of
+  Bedrock: i.e.: testing files in respective folders for template environments.
+
+## 3. Design Details
+
+This design seeks to introduce modular testing for terraform known as
+`Test Fixtures` based on best practices initially introduced by
+[Project Cobalt](github.com/microsoft/cobalt). The test fixtures decouples
+terraform commands to respective pipeline templats to be called and dynamically
+populated by a targeted template test.
+
+### 3.1 Embed new Infrastructure DevOps Model Flow - Continuous Integration
+
+Bedrock infrastructure integration tests have problematic gaps that do not
+account for terraform unit testing, state validation to live environments and
+staged release management for Bedrock versioning. In this design we wish to
+provide a single template leveraging MSI that verifies a new Infrastructure
+Testing Workflow that improves on the current Bedrock test harness.
+
+This design is intended to address expected core testing functionality
+including:
+
+- Support deployment of application-hosting infrastructure that will eventually
+  house the actual application service components Capture basic metrics and
+  telemetry from the deployment process for monitoring of ongoing pipeline
+  performance and diagnosis of any deployment failures
+- Support deployment into multiple staging environments
+- Execute automated unit-level and integration-level tests against the
+  resources, prior to deployment into any long-living environments
```

Comment:
Will we run integration tests against the production environment as well? If so, how would those tests differ form the ones run in the non-production environment.

Reply:
Yes our acceptance test on a later stage does a health check or perf test after an `apply` with the incoming feature. This can be a mock data manipulation, benchmark testing, or other verifications